pkg/network/cni/cni.go

```diff
@@ -203,17 +207,30 @@ func (plugin *cniNetworkPlugin) RemoveContainerNetwork(containerID string) error
 		return nil
 	}
 
-	// get the amount of combinations between an IP mask, and an iptables chain, with the specified container ID
-	// this makes the defaultNetworkName CNI network config not leak iptables rules
+	// Get the amount of combinations between an IP mask, and an iptables chain, with the specified container ID
+	// This makes the defaultNetworkName CNI network config not leak iptables rules
+	// It could possibly help with rule cleanup for other CNI network configs as well
 	result, err := getIPChains(c.ID)
 	if err != nil {
 		return err
 	}
-	comment := utils.FormatComment(defaultNetworkName, c.ID)
 
-	for _, t := range result {
-		if err = ip.TeardownIPMasq(t.ip, t.chain, comment); err != nil {
-			return err
+	for _, net := range plugin.cniConfig.Networks {
+		var hasBridge bool
+		for _, plugin := range net.Config.Plugins {
+			if plugin.Network.Type == "bridge" {
+				hasBridge = true
+			}
+		}
+
+		if hasBridge {
+			log.Debugf("TeardownIPMasq for container %q on CNI network %q which contains a bridge", containerID, net.Config.Name)
+			comment := utils.FormatComment(net.Config.Name, c.ID)
+			for _, t := range result {
+				if err = ip.TeardownIPMasq(t.ip, t.chain, comment); err != nil {
+					return err
```

Comment:
is it ok to do the immediate return here when fail?

or continue and collect the error in to an array, then check if the error size > 0?

I'm not sure which way is better because we're cleaning things here.

Reply:
I'll wrap the list of errs and then Errorf